source/common/filesystem/win32/filesystem_impl.cc

```diff
@@ -0,0 +1,105 @@
+#include <fcntl.h>
+#include <io.h>
+#include <sys/stat.h>
+#include <windows.h>
+
+#include "common/common/assert.h"
+#include "common/filesystem/filesystem_impl.h"
+
+#undef DELETE
+#undef GetMessage
+
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+#include "envoy/common/exception.h"
+
+#include "common/common/fmt.h"
+
+namespace Envoy {
+namespace Filesystem {
+
+bool InstanceImplWin32::fileExists(const std::string& path) {
+  const DWORD attributes = ::GetFileAttributes(path.c_str());
+  return attributes != INVALID_FILE_ATTRIBUTES;
+}
+
+bool InstanceImplWin32::directoryExists(const std::string& path) {
+  const DWORD attributes = ::GetFileAttributes(path.c_str());
+  if (attributes == INVALID_FILE_ATTRIBUTES) {
+    return false;
+  }
+  return attributes & FILE_ATTRIBUTE_DIRECTORY;
+}
+
+ssize_t InstanceImplWin32::fileSize(const std::string& path) {
+  struct _stat info;
+  if (::_stat(path.c_str(), &info) != 0) {
+    return -1;
+  }
+  return info.st_size;
+}
+
+std::string InstanceImplWin32::fileReadToEnd(const std::string& path) {
```

Comment:
some functions like this is same as the one in posix, extract?

Reply:
This one is almost the same -- the difference is that the file needs to be opened with `std::ios_base_binary` on Windows. If I understand that option correctly, it won't have any effect on posix, so we can probably pull this one out